table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -11,25 +11,198 @@
+   */
+  public static final class BypassEntry {
+    @JsonProperty("tables")
+    private final Set<BypassTableEntry> mEntries;
+
+    /**
+     * @param entries set of {@link BypassTableEntry}s
+     */
+    @JsonCreator
+    public BypassEntry(@JsonProperty("tables") @Nullable Set<BypassTableEntry> entries) {
+      mEntries = entries == null ? new HashSet<>() : entries;
+    }
+
+    /**
+     * Converts to a {@link UdbBypassSpec} object.
+     *
+     * @return the {@link UdbBypassSpec} object
+     */
+    public UdbBypassSpec toUdbBypassSpec() {
+      Map<String, Set<String>> map = mEntries.stream().collect(
+          Collectors.toMap(BypassTableEntry::getTable, BypassTableEntry::getPartitions));
+      return new UdbBypassSpec(map);
+    }
+
+    /**
+     * @return tables bypassed
+     */
+    public Set<String> getBypassedTables() {
+      return mEntries.stream().map(BypassTableEntry::getTable).collect(Collectors.toSet());
+    }
+  }
+
+  /**
+   * Table to partitions mapping.
+   */
+  @JsonDeserialize(using = BypassTableEntryDeserializer.class)
+  public static class BypassTableEntry {
+    private final String mTableName;
+    private final Set<String> mPartitions;
+
+    /**
+     * @param tableName table name
+     * @param partitions partition names
+     */
+    @JsonCreator
+    public BypassTableEntry(@JsonProperty("table") String tableName,
+                            @JsonProperty("partitions") Set<String> partitions) {
+      Preconditions.checkArgument(!tableName.isEmpty(), "empty table name");
+      mTableName = tableName;
+      mPartitions = partitions;
+    }
+
+    /**
+     * @return table name
+     */
+    public String getTable() {
+      return mTableName;
+    }
+
+    /**
+     * @return partition names
+     */
+    public Set<String> getPartitions() {
+      return mPartitions;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      if (this == other) {
+        return true;
+      } else if (other == null) {
+        return false;
+      } else if (getClass() != other.getClass()) {
+        return false;
+      }
+      BypassTableEntry entry = (BypassTableEntry) other;
+      return Objects.equals(mTableName, entry.mTableName);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(mTableName);
+    }
   }
 
   /**
-   * @return the bypass set
+   * Deserializer of BypassTableEntry
+   *
+   * Enables flexible syntax: either a single table name can be specified, and all belonging
+   * partitions will be bypassed;
+   * or an object of form
+   * {"table": "tableName", "partitions": ["part1", "part2"]}
+   * can be used, and individual partitions can be specified.
    */
-  public Set<String> getBypassSet() {
-    return mBypassSet;
+  public static class BypassTableEntryDeserializer extends JsonDeserializer<BypassTableEntry> {
+    @Override
+    public BypassTableEntry deserialize(JsonParser jp, DeserializationContext cxt)
```

Comment:
add a unit test to cover this function.

Reply:
`BypassTableEntry` is registered to use this class to deserialize. In `DbConfigTest` I added server deserailization tests to test this function.